model/buildlogger_test.go

```diff
@@ -759,48 +789,45 @@ func TestBuildloggerCreateFindQuery(t *testing.T) {
 		assert.False(t, ok)
 		assert.Equal(t, true, search[bsonutil.GetDottedKeyName(logInfoKey, logInfoMainlineKey)])
 	})
-	t.Run("EmptyInfo", func(t *testing.T) {
-		opts.Info = LogInfo{}
+	t.Run("OmitFields", func(t *testing.T) {
+		opts.EmptyTestName = true
+		opts.LatestExecution = true
 		search := createFindQuery(opts)
-		assert.Equal(t, search[logCreatedAtKey], bson.M{"$lte": opts.TimeRange.EndAt})
-		assert.Equal(t, search[logCompletedAtKey], bson.M{"$gte": opts.TimeRange.StartAt})
-		assert.Equal(t, true, search[bsonutil.GetDottedKeyName(logInfoKey, logInfoMainlineKey)])
-		assert.Len(t, search, 3)
+		assert.Equal(t, opts.Info.Project, search[bsonutil.GetDottedKeyName(logInfoKey, logInfoProjectKey)])
+		assert.Equal(t, opts.Info.Version, search[bsonutil.GetDottedKeyName(logInfoKey, logInfoVersionKey)])
+		assert.Equal(t, opts.Info.Variant, search[bsonutil.GetDottedKeyName(logInfoKey, logInfoVariantKey)])
+		assert.Equal(t, opts.Info.TaskName, search[bsonutil.GetDottedKeyName(logInfoKey, logInfoTaskNameKey)])
+		assert.Equal(t, nil, search[bsonutil.GetDottedKeyName(logInfoKey, logInfoTestNameKey)])
+		assert.Equal(t, opts.Info.Trial, search[bsonutil.GetDottedKeyName(logInfoKey, logInfoTrialKey)])
+		assert.Equal(t, opts.Info.ProcessName, search[bsonutil.GetDottedKeyName(logInfoKey, logInfoProcessNameKey)])
+		assert.Equal(t, opts.Info.Format, search[bsonutil.GetDottedKeyName(logInfoKey, logInfoFormatKey)])
+		assert.Equal(t, bson.M{"$in": opts.Info.Tags}, search[bsonutil.GetDottedKeyName(logInfoKey, logInfoTagsKey)])
+		query, ok := search[bsonutil.GetDottedKeyName(logInfoKey, logInfoArgumentsKey)]
+		require.True(t, ok)
+		args := query.(bson.M)["$in"].([]bson.M)
+		for _, arg := range args {
+			count := 0
+			for key, val := range arg {
+				assert.Equal(t, opts.Info.Arguments[key], val)
+				assert.Zero(t, count)
```

Comment:
If you want the loop to have exactly 0 or 1 element, can you just require that `len(args) < 2` rather than have the counter expect at most one iteration?